physicalplan/plan.go

```diff
@@ -151,10 +151,19 @@ func newScalarBinaryOperator(e *parser.BinaryExpr, storage storage.Queryable, mi
 		return nil, err
 	}
 
-	if e.LHS.Type() == parser.ValueTypeScalar {
-		return binary.NewScalar(model.NewVectorPool(stepsBatch), rhs, lhs, e.Op, true)
+	var (
+		scalarSideLeft bool
+		bothScalar     bool
```

Comment:
Maybe we can use an enum for this instead?

Reply:
Done